pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3651,6 +3666,21 @@ func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, ser
 	return cloudprovider.DefaultLoadBalancerName(service)
 }
 
+// GetAWSLoadBalancerName is a specific implementation of GetLoadBalancerName that could return an error
+func (c *Cloud) GetAWSLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) (string, error) {
```

Comment:
This implementation seems fine to me. I'd get a review from someone on the aws team to be sure.

Reply:
thanks

Reply:
The thing that confuses me about this implementation is that we've replaced GetLoadBalancerName with something that does what we want but no longer matches the interface.  Is that ok?